src/broadcast/mpris/mediaplayer2.cpp

```diff
@@ -0,0 +1,72 @@
+#include "broadcast/mpris/mediaplayer2.h"
+
+#include <QApplication>
+#include <QDebug>
+#include <QString>
+#include <QWidget>
+
+#include "mixxx.h"
+#include "moc_mediaplayer2.cpp"
+#include "sources/soundsourceproxy.h"
+
+MediaPlayer2::MediaPlayer2(MixxxMainWindow* pMixxx, QObject* parent)
+        : QDBusAbstractAdaptor(parent),
+          m_pMixxx(pMixxx) {
+}
+
+bool MediaPlayer2::canQuit() const {
+    return true;
+}
+
+bool MediaPlayer2::fullscreen() const {
+    return m_pMixxx->isFullScreen();
+}
+
+void MediaPlayer2::setFullscreen(bool fullscreen) {
+    m_pMixxx->slotViewFullScreen(fullscreen);
```

Comment:
The QWidget dependency can be removed from this class by emitting a signal here

Reply:
Yes, we can emit a signal here, the issue is that someone needs to connect the MediaPlayer2 signal to the window. 
This way we get mpris internals into the window class which is also a bad idea. 

In addition we have also: 
m_pMixxx->isFullScreen();
m_pMixxx->raise();

As info in the reverse direction. 

Form this point of view the solution is not bad. 
If we have one day another window technology we "just" need to implement a common interface for both technologies. 
For now the current solution is just fine, because we don't know the requirements of the other window system.

Reply:
I disagree. We know this conflicts with #3638. That IMO must be resolved before merging this.